src/utils/functionRuntimeUtils.ts

```diff
@@ -0,0 +1,71 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as opn from 'opn';
+import * as vscode from 'vscode';
+import { extensionPrefix } from '../../src/ProjectSettings';
+import { DialogResponses } from '../DialogResponses';
+import { localize } from '../localize';
+import { cpUtils } from './cpUtils';
+import { nodeUtils } from './nodeUtils';
+
+export namespace functionRuntimeUtils {
+
+    const runtimePackage: string = 'azure-functions-core-tools';
+
+    export async function validateFunctionRuntime(outputChannel: vscode.OutputChannel): Promise<void> {
+        const errorMessage: string | undefined = await getRuntimeValidationError(outputChannel);
+        if (errorMessage) {
+            await promptDocumentationAction(errorMessage);
+        }
+    }
+
+    async function getRuntimeValidationError(outputChannel: vscode.OutputChannel): Promise<string | undefined> {
+        try {
+            await cpUtils.executeCommand(outputChannel, undefined, 'func');
+        } catch (error) {
+            return localize(
+                'azFunc.noFunctionRuntime',
+                '[Azure Functions] Azure-functions-core-tools is not installed. Please install it otherwise Azure Functions extension may fail.'
+            );
+        }
+
+        try {
+            await nodeUtils.validateNpmInstalled(outputChannel);
+        } catch (error) {
+            return localize(
+                'azFunc.npmNotFound',
+                '[Azure Functions] Failed to check function runtime version caused by "npm" not in path, would you like to check the document for the prerequisites?'
+            );
+        }
+
+        const outdatedInfo: string = await cpUtils.executeCommand(outputChannel, undefined, 'npm', 'outdated', runtimePackage, '-g');
+        const runtimeDetail: string[] = outdatedInfo.slice(outdatedInfo.indexOf(runtimePackage)).trim().split(/\s+/);
+        if (runtimeDetail[1] !== runtimeDetail[2]) {
+            return localize(
+                'azFunc.outdatedFunctionRuntime',
+                '[Azure Functions] New Function Runtime version "{0}" found. Current is "{1}". Please update it otherwise Azure Functions extension may fail.',
```

Comment:
Suggestion: "Your installed version of the Azure Functions Core Tools ("{0}") does not match the latest ("{1}"). Please update for the best experience."